library/psa_crypto.c

```diff
@@ -4293,6 +4361,19 @@ psa_status_t psa_cipher_finish( psa_cipher_operation_t *operation,
         return( PSA_ERROR_BAD_STATE );
     }
 
+    if( operation->driver_in_use == 1 )
+    {
+        status = psa_driver_wrapper_cipher_finish( &operation->ctx.driver,
+                                                   output,
+                                                   output_size,
+                                                   output_length );
+        if( status != PSA_SUCCESS )
+            goto error;
+
+        (void) psa_cipher_abort( operation );
```

Comment:
In that case, I'll rewrite the error block to be an exit block instead.

Reply:
It's mostly academic because there aren't many reasons for `abort()` to fail, but:

* When calling `abort()` from a function that's already failing, return the status you were planning to return. This means that the return status from `abort()` is ignored.
* When calling `abort()` to clean up an object that's done its job before returning a success, return the status of `abort()`.

Reply:
Yes. See the rewritten version, it now handles both driver and software operations in the same way.